sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -425,8 +427,27 @@ def prepare_data_for_lloyd(X_t, init, tol, copy_x):
         dtype=compute_dtype,
     )
 
-    X_mean = (sum_axis1_kernel(X_t) / compute_dtype(n_samples))[:, 0]
-    X_mean_is_zeroed = (X_mean == 0).astype(int).sum() == len(X_mean)
+    elementwise_binary_divide_kernel = make_elementwise_binary_ops_1d_kernel(
+        n_features, _divide, max_work_group_size
+    )
+
+    sum_of_squares_kernel = make_sum_reduction_2d_axis1_kernel(
+        size0=n_features,
+        size1=None,
+        work_group_size=max_work_group_size,
+        device=device,
+        dtype=compute_dtype,
+    )
+
+    X_mean = sum_axis1_kernel(X_t)[:, 0]
+
+    divisor = dpt.full(
+        sh=(1,), fill_value=compute_dtype(n_samples), dtype=compute_dtype, device=device
+    )
```

Comment:
Can divisor be null?

Reply:
That would mean there's 0 samples, that is not possible since the input validation pipeline enforces at least one sample. So, no.